src/Query.php

```diff
@@ -428,7 +428,11 @@ public function count()
             $this->_execute();
         }
 
-        return $this->__resultSet->total();
+        if(is_a($this->__resultSet, '\Cake\Datasource\ResultSetInterface')) {
+            return $this->__resultSet->total();
+        } else {
+            return false;
```

Comment:
Returning 0 if something wrong happens during the query execution would let think it really did return 0 items. Is it something that we really want ? If the `execute()` method can return false, couldn't the `count()` method do the same ?